.github/workflows/one-click-hosting-release.yml

```
@@ -0,0 +1,138 @@
+name: One Click Hosting Release
+
+on:
+  workflow_dispatch:
+    inputs:
+      pre_release:
+        type: boolean
+        description: 'Pre-release?'
+        required: false
+    branches:
+      - main
+env:
+  CHANNEL: 'cloud'
+  CHANGELOG_FILE: 'temp-changelog.txt'
+
+jobs:
+  release:
+    runs-on: ubuntu-22.04
+    steps:
+      - name: checkout branch
+        uses: actions/checkout@v3
+        with:
+          ref: ${{ github.ref }}
+      - name: Get Configuration
+        id: configuration
+        uses: maork-elementor/json-to-variables@main
+        with:
+          filename: './.github/workflows/config.json'
+          prefix: config
+      - name: Test permissions
+        uses: ./.github/workflows/permissions
+        with:
+          ENVIRONMENT: ${{ env.config_deployment_environment }}
+          DEPLOYMENT_PERMITTED: ${{ env.config_deployment_permitted }}
+          DEPLOYMENT_REPOSITORY_OWNER: ${{ env.config_deployment_repository_owner }}
+      - name: Install Node.js 18.x
+        uses: actions/setup-node@v3
+        with:
+          node-version: 18.x
+      - name: Is pre-release?
```

Suggest splitting this to a Yaml conditional or possibly an external script that just returns 'prerelease' or an empty value to make this easier to read.. 

```suggestion
      - name: Is pre-release?
        id: is_pre_release
        if: ${{ github.event.inputs.pre_release }} == true
        steps:
          - run: |
              PRE_RELEASE="-prerelease"
              echo "This is a prerelease execution"
      - name: is_not_a_pre_release?
        if: ${{ github.event.inputs.pre_release }} != true
        steps:
          - run: |
              PRE_RELEASE=""
              echo "This is a release execution"
      - name: Copy Pre-Release variable
        run: echo "PRE_RELEASE=${PRE_RELEASE}" >> $GITHUB_ENV
```

---

Good suggestion.
I think it is shorter as it is now.